presto-main/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -49,6 +68,33 @@ public QuerySessionSupplier(
     }
 
     @Override
+    public void addSessionPropertyConfigurationManager(SessionPropertyConfigurationManagerFactory sessionConfigFactory)
+    {
+        if (sessionPropertyConfigurationManagerFactories.putIfAbsent(sessionConfigFactory.getName(), sessionConfigFactory) != null) {
+            throw new IllegalArgumentException(format("Session property configuration manager '%s' is already registered", sessionConfigFactory.getName()));
+        }
+    }
+
+    @Override
+    public void loadSessionPropertyConfigurationManager()
+            throws Exception
+    {
+        if (SESSION_PROPERTY_CONFIGURATION.exists()) {
+            Properties properties = new Properties();
```

Comment:
We have a `loadProperties()` utility method in several classes that do this. Create a commit before this one that consolidates all of those into a `PropertiesUtil` class in `com.facebook.presto.util`, then you can use that here.